javascript/react/basics-i/rendering-multiple-components.md

```diff
@@ -32,34 +32,35 @@ aspects:
 ---
 ## Content
 
-**React** allows the creation of `DOM` element collections by harnessing the power of pure `JS`.
+**React** allows the creation of `DOM` element collections by harnessing the power of pure JS.
 
-For example, you can dynamically generate a collection of `<p>` items by transforming an `array` with the `map()` function:
+For example, you can dynamically generate a collection of `<p>` items by transforming an array with the `map()` function:
 ```jsx
 const input = [0, 1, 2, 3];
-const collection = input.map((no) => {
-    return <p>{no}</p>;
-  });
+const collection = input.map(no => {
+  return <p>{no}</p>;
+});
 ```
 
-When rendering such lists inside a `component`, an additional **identity attribute** named `key` must be included for every element. `key`s should be unique to each **sibling element**.
+When rendering such lists inside a component, an additional **identity attribute** named `key` must be included for every element. `key`s should be unique to each **sibling element**.
```

Comment:
```suggestion
When rendering such lists inside a component, an additional **identity attribute** named `key` must be included for every element. `key`s should be unique among all sibling elements.
```

Also add a footnote explaining why? https://dev.to/jtonzing/the-significance-of-react-keys---a-visual-explanation--56l7

Reply:
added this footnote:

To get a better understanding of this, let's take a look at how React decides when it should render or re-render a component. Knowing that in its rawest form, a React component is just an object with a set of properties, each time a change occurs in these properties React will compare the new and old versions and, if a difference is detected, it will redraw the component. If the defined `key` is not unique, the order of the components might get mixed which will lead to React thinking that all components have changed (instead of just one). This, in turn, will lead to React re-rendering all the components (instead of just the one that was changed) which might lead to performance issues. Take a look at [this article](https://dev.to/jtonzing/the-significance-of-react-keys---a-visual-explanation--56l7) if you want to learn more about this.